### ISSUE TITLE
Apps aren't loaded yet.

### ISSUE THREAD

**Issue by YassWarrior** [justquick/django-activity-stream#279]
Hello, 

I try to use this app but I have a problem with the configuration.

The name of my app : main_app

apps.py of main_app : 

```
from django.apps import AppConfig
from actstream import registry

class MainAppConfig(AppConfig):
	name = 'main_app'

def ready(self):
	registry.register(self.get_model('Inscrit'))
```

__init__ of main_app : 
`
default_app_config = 'main_app.apps.MainAppConfig'`

settings.py [Installed_apps] :

```
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.humanize',
    'django.contrib.sites',
    'bootstrapform',
    'main_app',
    'actstream',
]
```
I have this error : 

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 350, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 342, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 176, in fetch_command
    commands = get_commands()
  File "/usr/local/lib/python2.7/dist-packages/django/utils/lru_cache.py", line 100, in wrapper
    result = user_function(*args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 71, in get_commands
    for app_config in reversed(list(apps.get_app_configs())):
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 137, in get_app_configs
    self.check_apps_ready()
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

I read the #272 but it doesn't help me.. 

BUT when I write from `actstream import registry` inside `def ready`, it works but why ? :octocat:

**Comment by dolugen** [comment]
I remember having this issue. I think I just re-ordered `INSTALLED_APPS` a bit.

**Comment by fatzh** [comment]
any clue on the expected order ? I got `actstream ` as the last app in my list and got the same error... everything is working fine without it.

**Comment by fatzh** [comment]
nevermind, I updated to 0.6.1 directly from git and it seems to work now :)

**Comment by Debetux** [comment]
Same issue here.

```
Django==1.9.2
django-activity-stream==0.6.1
django-annoying==0.8.7
django-crispy-forms==1.6.0
django-jsonfield==0.9.19
django-model-utils==2.4
Pillow==3.1.1
wheel==0.29.0
```

```
RuntimeError: Model class django.contrib.sites.models.Site doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

Trying to understand why...

**Comment by rkakrik** [comment]
Installed from git, but same problem((

**Comment by szechyjs** [comment]
installed from git as well.

```
Traceback (most recent call last):
  File "/Users/jared/code/services/venv/bin/manage.py", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/jared/code/services/bin/marrts.py", line 10, in <module>
    django.setup()
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/apps/config.py", line 116, in create
    mod = import_module(mod_path)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/jared/code/services/lost/conman/apps.py", line 2, in <module>
    from actstream import registry
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/django_activity_stream-0.6.1-py2.7.egg/actstream/registry.py", line 11, in <module>
    from actstream.compat import generic, get_model
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/django_activity_stream-0.6.1-py2.7.egg/actstream/compat.py", line 22, in <module>
    from django.contrib.contenttypes import fields as generic
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/contrib/contenttypes/fields.py", line 5, in <module>
    from django.contrib.contenttypes.models import ContentType
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/contrib/contenttypes/models.py", line 161, in <module>
    class ContentType(models.Model):
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/db/models/base.py", line 94, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/apps/registry.py", line 239, in get_containing_app_config
    self.check_apps_ready()
  File "/Users/jared/code/services/venv/lib/python2.7/site-packages/Django-1.9.3-py2.7.egg/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

**Comment by justquick** [comment]
@szechyjs hrm, something seems to be going wrong in your app config. can you post your conman/apps.py file?

**Comment by jbsag** [comment]
tl;dr: the band-aid is to do something like this in your apps.py

```class YourAppConfig(AppConfig):
    name = "yourapp"
    def ready(self):
        from actstream import registry
        registry.register(self.get_model('yourapp'))
```

Why? Because registry is not imported until ready() is called. If registry is imported beforehand, we see in @szechyjs 's stack trace that actstream.compact imports fields from django.contrib.contentypes. This module imports the model ContentType. See: https://docs.djangoproject.com/en/1.9/ref/applications/#how-applications-are-loaded . "In other words, your applications’ root packages and the modules that define your application configuration classes shouldn’t import any models, even indirectly."

To completely fix this issue, we have to somehow prevent any other models imported indirectly before we can use actstream's registry module.

**Comment by justquick** [comment]
yea that's the best approach. docs updated in #291

**Comment by kayatomic** [comment]
@justquick do you have any plans to add support for Django 1.10?
I'm currently using Django 1.10 and have tried @jbsag comment above but I'm still getting the exception **django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.**

**Comment by auvipy** [comment]
update your activity stream version

**Comment by kayatomic** [comment]
@auvipy according to the change log, author hasn't added support for Django 1.10 - [http://django-activity-stream.readthedocs.io/en/latest/changelog.html](url)

Were you able to get this working on Django 1.10?
